### DISCUSSION TITLE
Property parameter

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#597] [Q&A]
Hi everyone, How to get property's Parameters as `AsmResolver.DotNet.Collections.Parameter` or `ParameterDefinition ` from `PropertyDefinition ` ?, is there any way to do that? please provide sample code for demonstration.

Thanks

**Comment by Washi1337** [comment]
Properties do not define parameter definitions, only a property signature with the parameter types and return type (exposed under `PropertyDefinition::Signature`)

If you really need a `Parameter` or `ParameterDefinition`, you'd have to get them from the appropriate `GetMethod` or `SetMethod`.

```csharp
PropertyDefinition property = ...;

foreach (var parameter in property.GetMethod.Parameters)
{
   Console.WriteLine(parameter.Name);
}
```

**Reply by dashne** [reply]
thank you, but when we say Parameters of property, are they Parameters of Get Method or Set Method, or both?

**Reply by Washi1337** [reply]
Getters and setters are both required to have a specific signature, with the only difference between the two being that setters have the return value moved into an additional `value` parameter.

This means the getter method will the most representative for the parameters and return value of the property. However, while rare, not all properties will have getters, so you may have to fall back to the setter of the property.
